### PR TITLE
Fix 'path' argument. Add more visitor tests

### PR DESCRIPTION
**Pull request by IvanGoncharov**
As discussed in #1145 I started to work on tests for `visit` function.

### CODE REVIEW

**Comment by IvanGoncharov** [comment]
@leebyron I added the simple test for `path` parameter and discover that `path` value differs between `enter` and `leave` callbacks. On `leave` path is always one element shorter, for example if you was called on enter to `OperationDefinition` node `path` value would be `['definitions', 0]` but on leave you will get `['definitions']`.

Is it intentional behaviour or a bug?

File: src/language/__tests__/visitor-test.js [hunk 1]
```diff
@@ -17,6 +17,35 @@ import { testSchema } from '../../validation/__tests__/harness';
...
+      enter(node, key, parent, path) {
+        visited.push(['enter', path.slice()]);
+      },
+
+      leave(node, key, parent, path) {
+        visited.push(['leave', path.slice()]);
+      },
+    });
+
+    console.log(JSON.stringify(visited, null, 2));
```
Comment (mjmahone) [inline src/language/__tests__/visitor-test.js]: Probably don't want to keep this console.log in the test

File: src/language/__tests__/visitor-test.js [hunk 2]
```diff
@@ -17,6 +17,35 @@ import { testSchema } from '../../validation/__tests__/harness';
 import { getNamedType, isCompositeType } from '../../type';
 
 describe('Visitor', () => {
+  it('check path argument', () => {
```
Comment (mjmahone) [inline src/language/__tests__/visitor-test.js]: nit: 'validates path argument' or 'checks path argument'

Comment (IvanGoncharov) [reply src/language/__tests__/visitor-test.js]: @mjmahone Thanks 👍 Fixed now.

File: src/language/__tests__/visitor-test.js (hunk 1 above)
Comment (IvanGoncharov) [reply src/language/__tests__/visitor-test.js]: @mjmahone Consequences of committing at 3 AM 😄 
Removed.

**Comment by IvanGoncharov** [comment]
I fixed this with `path` by a very small fix that's easy to verify.

As for testing visitor callback arguments, I decided it's too much to add it hardcoded snapshots inside tests. So I added `checkVisitorFnArgs` utility function that checks types and interdependencies of all callback arguments and calls it from every callback inside `visitor-tests.js`.

One thing to note is the current implementation of `visit` doesn't maintain consistency of arguments after AST is edited. For example, if you edit some node on `enter` you would get updated node on `leave` but `parent` and all ancestors will remain not updated. To workaround this problem I added `isEdited` argument that disables some of the checks inside `checkVisitorFnArgs` function.

@mjmahone @leebyron Can you please review this PR?

**Comment by leebyron** [comment]
Nice work, @IvanGoncharov

**Pull request merged**